packages/persist/lib/controllers/persist.controller.ts

```diff
@@ -117,17 +110,10 @@ class PersistController {
         } = req;
         const logCtx = logContextGetter.get({ id: String(activityLogId) });
         const persist = async (records: FormattedRecord[], legacyRecords: DataRecord[]) => {
-            recordsService
-                .upsert(records, nangoConnectionId, model, true)
-                .then((res) => {
-                    if (isErr(res)) {
-                        throw res.err;
-                    }
-                })
-                .catch((err: unknown) => {
-                    logger.error(`Failed to delete records: ${stringifyError(err)}`);
-                });
-            return await dataService.upsert(legacyRecords, nangoConnectionId, model, activityLogId, environmentId, true, logCtx);
+            const newUpsert = recordsService.upsert(records, nangoConnectionId, model, true);
```

Comment:
Unclear what the `true` is doing here. Feel like this is left over from code I probably wrote...

Reply:
it is `softDelete`. Passing a struct with named field would be better. I will make a PR

Reply:
addressed in https://github.com/NangoHQ/nango/pull/2080